src/DynamoCoreWpf/UI/InOutPortPanel.cs

```diff
@@ -48,8 +54,14 @@ protected override Size MeasureOverride(Size constraint)
                     // Default behavior of getting each child to measure.
                     child.Measure(constraint);
                 }
-                catch(System.Exception)
+                catch(XamlParseException e)
                 {
+                    if(e != null)
+                     //aggregate the stack traces as we'll loop over all children in this panel.
+                    {
+                        aggretatedExceptionStack += $"------------------ {System.Environment.NewLine } { e.StackTrace} {System.Environment.NewLine}" +
+                        e.InnerException != null ? $" inner: {e.InnerException.StackTrace} {System.Environment.NewLine}" : string.Empty;
+                    }
```

Comment:
what about 
```
List<XamlParseException> exList = null;

try
{
    // Default behavior of getting each child to measure.
    child.Measure(constraint);
}
catch(XamlParseException e)
{
    exList.Add(e);
    continue;
}

//log to analytics if we recorded any exceptions
if(exList.Count > 0)
{
    Analytics.TrackException(new AggregateException("MeasureOverride XML exception", exList), false);
}
```
Then, In analytics.Core, you can simply handle the case when the Exception is of type aggregate
```
        public ExceptionEvent(Exception ex, bool isFatal = true)
        {
            exception = ex;
            Description = ex.GetType().ToString() + Environment.NewLine;
            if (!string.IsNullOrEmpty(ex.StackTrace))
                Description += ex.StackTrace.ToString().Replace("&", "");

            if (exception is AggregateException aggrEx)
            {
                foreach (var iex in aggrEx.Flatten().InnerExceptions) {
                    Description += "\n Internal exception \n" + iex.StackTrace.ToString().Replace("&", "");
                }
            }
            
            IsFatal = isFatal;
        }
 ```

Reply:
Unless you are trying to avoid modifying the Analytics assembly

Reply:
well yes, I was trying to avoid altering the analytics dll - but more importantly it doesn't really handle @aparajit-pratap 's use case here:
https://github.com/DynamoDS/Dynamo/pull/12395

I guess he could make an AggregateException where the inner exception is just another exception of the new aggregate... but feels like a bunch of incantations just to get a string in the end.

I'm okay with it if y'all feel it's more idiomatic and makes the API easier to understand?

Reply:
It does nicely avoid a new exception type though.

Reply:
I think using AggregateException would be simpler
And I think we should enhance the existing Analtics.Core TrackException API to handle aggregates, to use the inner exceptions (if they exist) and to use the Exception message (not just the stackTrace)